### ISSUE TITLE
TextField accessibility on Linux

### ISSUE THREAD

**Issue by RastislavKish** [flutter/flutter#113049]
On Linux, TextField has some accessibility issues.

## Steps to reproduce

1. Run the sample code below
2. Start Orca screenreader, on systems where it's preinstalled usually by Super+alt+S, or simply by launching orca
3. In the application window, try to type something in the TextField
4. Navigate around the field using arrow keys

## Expected behavior

When navigating character by character (by left and right arrow keys), Orca should read the currently focused-one.
Up/Down arrow keys (navigation by lines) should cause Orca to read the focused line.

## Actual behavior

Orca reports "blank" for almost any kind of movement, except navigation by words (Ctrl+Left & Ctrl+Right), which surprisingly works as it should.

## Environment

Ubuntu Mate 22.04 64-bit\
Orca 43.1

## Sample code

```
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(
    MaterialApp(
        home: Column(
            children: <Widget>[
                TextField(
                    decoration: InputDecoration(
                        labelText: 'Input some text',
                        ),
                    ),
                ElevatedButton(
                    child: const Text('Close'),
                    onPressed: () => SystemNavigator.pop(),
                    ),
                ],
            )
        ),
    );
```

## Flutter doctor -v output

[✓] Flutter (Channel master, 3.5.0-2.0.pre.23, on Ubuntu 22.04.1 LTS 5.15.0-48-generic, locale sk_SK.UTF-8)
    • Flutter version 3.5.0-2.0.pre.23 on channel master at /home/username/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision be86a91654 (76 minutes ago), 2022-10-06 09:35:08 -0700
    • Engine revision e72434025f
    • Dart version 2.19.0 (build 2.19.0-283.0.dev)
    • DevTools version 2.18.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/username/Android/Sdk
    • Platform android-31, build-tools 30.0.3
    • ANDROID_SDK_ROOT = /home/username/Android/Sdk
    • Java binary at: /usr/lib/jvm/java-11-openjdk-amd64/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.16+8-post-Ubuntu-0ubuntu122.04)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] Connected device (1 available)
    • Linux (desktop) • linux • linux-x64 • Ubuntu 22.04.1 LTS 5.15.0-48-generic

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.

**Comment by huycozy** [comment]
@RastislavKish thanks for filing the issue. This issue is reproducible on the latest `stable` and `master` channels.

<details>
<summary>Demo video</summary>

https://user-images.githubusercontent.com/104349824/194524206-c4a85fc9-1608-4306-a254-fe69cdd1c24a.mp4

</details>

<details>
<summary>Sample code (modified sample code from OP, to be runnable)</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(
      MaterialApp(
        home: Scaffold(
          body: Column(
            children: <Widget>[
              TextField(
                decoration: InputDecoration(
                  labelText: 'Input some text',
                ),
              ),
              ElevatedButton(
                child: const Text('Close'),
                onPressed: () => SystemNavigator.pop(),
              ),
            ],
          ),
        ),
      ),
    );
```
</details>

<details>
<summary>flutter doctor -v (stable & master)</summary>

```bash
[✓] Flutter (Channel stable, 3.3.4, on Ubuntu 22.04.1 LTS 5.15.0-48-generic, locale en_US.UTF-8)
    • Flutter version 3.3.4 on channel stable at /home/huynq/Documents/Working/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision eb6d86ee27 (27 hours ago), 2022-10-04 22:31:45 -0700
    • Engine revision c08d7d5efc
    • Dart version 2.18.2
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /home/huynq/Android/Sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /home/huynq/Documents/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.22.1
    • ninja version 1.10.1
    • pkg-config version 0.29.2

[✓] Android Studio (version 2021.1)
    • Android Studio at /home/huynq/Documents/android-studio
    • Flutter plugin version 67.0.1
    • Dart plugin version 211.7817
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] VS Code (version 1.71.0)
    • VS Code at /usr/share/code
    • Flutter extension version 3.48.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 22.04.1 LTS 5.15.0-48-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 105.0.5195.102

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```bash
[✓] Flutter (Channel master, 3.5.0-2.0.pre.30, on Ubuntu 22.04.1 LTS 5.15.0-48-generic, locale
    en_US.UTF-8)
    • Flutter version 3.5.0-2.0.pre.30 on channel master at /home/huynq/Documents/Working/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4536bbbd46 (27 minutes ago), 2022-10-06 14:05:22 -0400
    • Engine revision 14d18e7b10
    • Dart version 2.19.0 (build 2.19.0-283.0.dev)
    • DevTools version 2.18.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /home/huynq/Android/Sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /home/huynq/Documents/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.22.1
    • ninja version 1.10.1
    • pkg-config version 0.29.2

[✓] Android Studio (version 2021.1)
    • Android Studio at /home/huynq/Documents/android-studio
    • Flutter plugin version 67.0.1
    • Dart plugin version 211.7817
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] VS Code (version 1.71.0)
    • VS Code at /usr/share/code
    • Flutter extension version 3.50.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 22.04.1 LTS 5.15.0-48-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 105.0.5195.102

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```
</details>

**Comment by jpnurmi** [comment]
`FlAccessibleTextField` is missing implementation for some `AtkText` methods, notably `get_text_at/before/after_offset()`. We need a way to detect char, word, sentence, and line boundaries. GTK does that with help of Pango.

https://github.com/flutter/engine/blob/main/shell/platform/linux/fl_accessible_text_field.cc#L348
